collectors/proc.plugin/sys_fs_btrfs.c

```diff
@@ -736,6 +1019,177 @@ int do_sys_fs_btrfs(int update_every, usec_t dt) {
+            rrddim_set_by_pointer(node->st_commit_timings, node->rd_commit_timings_max, node->commit_timings_max);
+            rrdset_done(node->st_commit_timings);
+        }
+
+        // --------------------------------------------------------------------
+        // error_stats per device
+
+        if(do_error_stats == CONFIG_BOOLEAN_YES ||  (do_error_stats == CONFIG_BOOLEAN_AUTO &&
+                                                    (node->devices ||
+                                                    netdata_zero_metrics_enabled == CONFIG_BOOLEAN_YES))) {
+            do_error_stats = CONFIG_BOOLEAN_YES;
+
+            for(BTRFS_DEVICE *d = node->devices ; d ; d = d->next) {
+
+                if(unlikely(!d->st_error_stats)) {
+                    char id[RRD_ID_LENGTH_MAX + 1], name[RRD_ID_LENGTH_MAX + 1], title[200 + 1];
+
+                    snprintfz(id, RRD_ID_LENGTH_MAX, "error_stats_%s_dev%d", node->id, d->id);
+                    snprintfz(name, RRD_ID_LENGTH_MAX, "error_stats_%s_dev%d", node->label, d->id);
+                    snprintfz(title, 200, "BTRFS Errors");
+
+                    netdata_fix_chart_id(id);
+                    netdata_fix_chart_name(name);
+
+                    d->st_error_stats = rrdset_create_localhost(
+                            "btrfs"
+                            , id
+                            , name
+                            , node->label
+                            , "btrfs.errors"
+                            , title
+                            , "errors"
+                            , PLUGIN_PROC_NAME
+                            , PLUGIN_PROC_MODULE_BTRFS_NAME
+                            , NETDATA_CHART_PRIO_BTRFS_ERRORS
+                            , update_every
+                            , RRDSET_TYPE_LINE
+                    );
+
+                    char rd_id[RRD_ID_LENGTH_MAX + 1];
+                    snprintfz(rd_id, RRD_ID_LENGTH_MAX, "write_errs");
+                    d->rd_write_errs = rrddim_add(d->st_error_stats, rd_id, NULL, 1, 1, RRD_ALGORITHM_ABSOLUTE);
+                    snprintfz(rd_id, RRD_ID_LENGTH_MAX, "read_errs");
+                    d->rd_read_errs = rrddim_add(d->st_error_stats, rd_id, NULL, 1, 1, RRD_ALGORITHM_ABSOLUTE);
+                    snprintfz(rd_id, RRD_ID_LENGTH_MAX, "flush_errs");
+                    d->rd_flush_errs = rrddim_add(d->st_error_stats, rd_id, NULL, 1, 1, RRD_ALGORITHM_ABSOLUTE);
+                    snprintfz(rd_id, RRD_ID_LENGTH_MAX, "corruption_errs");
+                    d->rd_corruption_errs = rrddim_add(d->st_error_stats, rd_id, NULL, 1, 1, RRD_ALGORITHM_ABSOLUTE);
+                    snprintfz(rd_id, RRD_ID_LENGTH_MAX, "generation_errs");
+                    d->rd_generation_errs = rrddim_add(d->st_error_stats, rd_id, NULL, 1, 1, RRD_ALGORITHM_ABSOLUTE);
+
+                    add_labels_to_btrfs(node, d->st_error_stats);
```

Comment:
Device error should have a `device_id` label too.

Reply:
It is not the same because `instance` is the chart id (the whole thing with filesystem uuid).